v-next/hardhat/src/internal/core/hre.ts

```diff
@@ -134,7 +134,14 @@ export class HardhatRuntimeEnvironmentImplementation
       globalOptionDefinitions,
     );
 
-    await hooks.runSequentialHandlers("hre", "created", [hre]);
```

Comment:
I switched this as the mock artifact plugin (which depends on the artifact plugin) was being run before the artifacts plugin. This is because `runSequentialHandlers` runs in reverse priority order.

I am unsure if this is the right move. But plugin authors who enhance the `hre` are likely to want to execute after other `hre` enhancements they are depending on.

Reply:
This change has been dropped. We fixed the underlying bug with the `runSequentialHandlers` order.